src/utils/state/Done.hpp

```diff
@@ -0,0 +1,25 @@
+#pragma once
+#ifndef DONE_HPP
+#define DONE_HPP
+
+#include <utils/state/IState.hpp>
+
+/* ************************************************************************** */
+template<typename Context>
+class Done : public IState<Context>
+{
+public:
+  explicit Done(Context* ptr);
+  ~Done();
```

Comment:
Starting from C++11 it is possible to `= default` special member functions (default ctor, copy functions, move functions, dtor). This way they still get generated from the compiler even if they usually wouldn't (f.e. no default ctor if any other ctor is defined).
As far as I know, it's best practice to always do that if possible. It's especially useful when declaring a destructor just to make it virtual.

If you'd like we can practice that habit here and define the destructor that does nothing in the header file, like this:
```suggestion
  ~Done() {}
```